ckanext/unaids/tests/test_plugin.py

```diff
@@ -30,3 +33,94 @@ def test_geojson_format_guessed_correctly(self):
+        _update_resource_last_modified_date(resource_with_file)
+        assert 'last_modified' in resource_with_file
+
+    def test_file_to_null_should_not_update_last_modified_datetime(self, resource_with_file):
+        null_resource = {}
+        _update_resource_last_modified_date(
+            null_resource, current=resource_with_file
+        )
+        assert 'last_modified' not in resource_with_file
+
+    def test_file_to_file_should_update_last_modified_datetime(self, resource_with_file):
+        updated_resource = resource_with_file.copy()
+        updated_resource.update({'url': 'file2.csv'})
+        _update_resource_last_modified_date(
+            updated_resource, current=resource_with_file
+        )
+        assert 'last_modified' in updated_resource
+
+    def test_file_to_link_should_update_last_modified_datetime(self, resource_with_file, resource_with_link):
+        _update_resource_last_modified_date(
+            resource_with_link, current=resource_with_file
+        )
+        assert 'last_modified' in resource_with_link
+
+    def test_changing_metadata_in_file_should_not_update_last_modified_datetime(self, resource_with_file):
+        updated_resource = resource_with_file.copy()
+        updated_resource.update({'key': 'value'})
+        _update_resource_last_modified_date(
+            resource_with_file, current=resource_with_file
+        )
+        assert 'last_modified' not in updated_resource
+
+    def test_null_to_link_should_update_last_modified_datetime(self, resource_with_link):
+        _update_resource_last_modified_date(resource_with_link)
+        assert 'last_modified' in resource_with_link
+
+    def test_link_to_null_should_not_update_last_modified_datetime(self, resource_with_link):
+        null_resource = {}
+        _update_resource_last_modified_date(
+            null_resource, current=resource_with_link
+        )
+        assert 'last_modified' not in resource_with_link
+
+    def test_link_to_link_should_update_last_modified_datetime(self, resource_with_link):
+        updated_resource = resource_with_link.copy()
+        updated_resource.update({'url': 'http://example2.com'})
+        _update_resource_last_modified_date(
+            updated_resource, current=resource_with_link
+        )
+        assert 'last_modified' in updated_resource
+
+    def test_link_to_file_should_update_last_modified_datetime(self, resource_with_file, resource_with_link):
+        _update_resource_last_modified_date(
+            resource_with_file, current=resource_with_link
+        )
+        assert 'last_modified' in resource_with_file
+
+    def test_changing_metadata_in_link_should_not_update_last_modified_datetime(self, resource_with_link):
+        updated_resource = resource_with_link.copy()
+        updated_resource.update({'key': 'value'})
+        _update_resource_last_modified_date(
+            resource_with_link, current=resource_with_link
```

Comment:
![image](https://user-images.githubusercontent.com/30174560/124472906-f92d5100-dd9e-11eb-9e3b-2bf90ccc2344.png)
Update the line with `updated_resource`